### PR TITLE
Simplify install.ps1 and move core module files under src/

### PR DESCRIPTION
**Pull request by rkeithhill**
This moves the core module files under a src dir.  BTW can we move install.ps1 to the chocolatey dir?  Shipping this file in the module may lead to confusion.  Also both it and CheckVersion.ps1 check if git is installed.

### CODE REVIEW

File: test/Utils.Tests.ps1 [hunk 1]
```diff
@@ -21,7 +22,7 @@ Describe 'Utils Function Tests' {
             Get-FileEncoding $profilePath | Should Be 'utf8'
             $content = Get-Content $profilePath
             $content.Count | Should Be 2
-            $modulePath = Resolve-Path $PSScriptRoot\..
+            # $modulePath = Resolve-Path $PSScriptRoot\..
```
Comment (dahlbyk) [inline test/Utils.Tests.ps1]: Delete?

**Comment by rkeithhill** [comment]
Will this cause merge headaches for the outstanding PRs?

**Comment by dahlbyk** [comment]
> Will this cause merge headaches for the outstanding PRs?

Kind of. I wasn't comfortable with the overlap between this and #358, so I've overwritten your changes here based on the non-Chocolatey changes that ended up over there. Net result:

- `profile.example.ps1` has been moved back up to the project root, to avoid breaking those who have used `install.ps1` or Chocolatey until now.
- I also added a warning to that file indicating it will be removed in a future version (presumably 1.0).
- `install.ps1` no longer uses `profile.example.ps1`, instead calling `Import-Module` directly.
  - This is kind of a breaking change, as fresh installs will no longer call `Start-SshAgent -Quiet`. Thoughts? (cc #338)
- Finally, the module files are moved under `src/` along with some missing fixes for references to those files.

Note: there are currently a few commits in this branch marked as `fixup!`. If you see that, it should be a clear indication that the branch is meant to be `rebase -i`'d before it's merged (autosquash FTW).

**Comment by rkeithhill** [comment]
So should I run `git rebase -i master` on this branch and then `git push --force-with-lease` before merging it?

**Comment by dahlbyk** [comment]
> So should I run `git rebase -i master` on this branch and then `git push --force-with-lease` before merging it?

Yep, looks good as merged.

**Pull request merged**